File: VB/FilterCriteriaSample/DataModel/ProductAggregate.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace FilterCriteriaSample
    Friend Class ProductAggregate
        Public Property ProductName() As String
        Public Property CategoryName() As String

        <StringLength(20)> _
        Public Property QuantityPerUnit() As String

        Public Property UnitPrice() As Decimal

        Public Property UnitsInStock() As Short

        Public Property UnitsOnOrder() As Short

        Public ReadOnly Property OnOrderIncome() As Decimal
            Get
                Return UnitsOnOrder * UnitPrice
            End Get
        End Property
    End Class
End Namespace

File: CS/FilterCriteriaSample/DataModel/ProductAggregate.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FilterCriteriaSample {
    class ProductAggregate {
        public string ProductName { get; set; }
        public string CategoryName { get; set; }

        [StringLength(20)]
        public string QuantityPerUnit { get; set; }

        public decimal UnitPrice { get; set; }

        public short UnitsInStock { get; set; }

        public short UnitsOnOrder { get; set; }

        public decimal OnOrderIncome {
            get {
                return UnitsOnOrder * UnitPrice;
            }
        }
    }
}
